e2e/LongHaul/service/IotHub.cs

```diff
@@ -52,32 +52,8 @@ public async Task RunAsync(CancellationToken ct)
 
             while (!ct.IsCancellationRequested)
             {
-                var payload = new CustomDirectMethodPayload
-                {
-                    RandomId = Guid.NewGuid(),
-                    CurrentTimeUtc = DateTimeOffset.UtcNow,
-                    MethodCallsCount = ++methodCallsCount,
-                };
-
-                var methodInvocation = new DirectMethodServiceRequest("EchoPayload")
-                {
-                    Payload = payload,
-                    ResponseTimeout = TimeSpan.FromSeconds(30),
-                };
-
-                _logger.Trace($"Invoking direct method for device: {_deviceId}", TraceSeverity.Information);
-
-                // Invoke the direct method asynchronously and get the response from the simulated device.
-                DirectMethodClientResponse response = await s_serviceClient.DirectMethods.InvokeAsync(_deviceId, methodInvocation);
-
-                if (response.TryGetPayload(out CustomDirectMethodPayload responsePayload))
-                {
-                    _logger.Metric(
-                        D2cDirectMethodDelaySeconds,
-                        (DateTimeOffset.UtcNow - responsePayload.CurrentTimeUtc).TotalSeconds);
-                }
-
-                _logger.Trace($"Response status: {response.Status}, payload:\n\t{JsonConvert.SerializeObject(response.PayloadAsString)}", TraceSeverity.Information);
+                await InvokeDirectMethodAsync(++methodCallsCount, ct).ConfigureAwait(false);
+                await SetDesiredPropertiesAsync("methodCallsCount", methodCallsCount.ToString(), ct).ConfigureAwait(false);
```

Comment:
I am still calling both methods from `RunAsync()` here as we need to pass in the `methodCallsCount` into both methods and increase its value per each loop.

Reply:
Let's just send a guid as a string and not tie the operations together.